src/transforms/select.jl

```diff
@@ -2,6 +2,38 @@
 # Licensed under the MIT License. See LICENSE in the project root.
 # ------------------------------------------------------------------
 
+struct TableSelection{T} 
+  table::T
+  cols::Vector{Symbol}
+end
+
+Tables.istable(::Type{<:TableSelection}) = true
+Tables.columnaccess(::Type{<:TableSelection}) = true
+Tables.columns(ts::TableSelection) = ts
+Tables.columnnames(ts::TableSelection) = ts.cols
+Tables.getcolumn(ts::TableSelection, col::Int) =
+  Tables.getcolumn(ts.table, ts.cols[col])
+Tables.getcolumn(ts::TableSelection, col::Symbol) = 
+  Tables.getcolumn(ts.table, col)
+
+function Tables.schema(ts::TableSelection)
+  schema = Tables.schema(ts.table)
+  tbtypes = schema.types
+  tbnames = collect(schema.names)
+  selecttypes = tbtypes[indexin(ts.cols, tbnames)]
+  return Tables.Schema(ts.cols, selecttypes)
+end
+
+Tables.materializer(::Type{TableSelection{T}}) where {T} = 
+  Tables.materializer(T)
```

Comment:
Can you please confirm that the materializer of the parent type is what we want here? I thought that the materializer function had to return the constructor of the type in question, in this case a TableSelection. But it may make sense to return the constructor of the parent given that the selection is a subtable always? We need to double check the API docs.

Reply:
Resolved in commit af43f28.